src/averageAges.js

```diff
@@ -55,7 +73,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = !onlyWithSon
+    ? people.filter(child => (
+      people.find(person => child.mother === person.name)
+    ))
+    : people.filter(child => (
+      people.find(person => child.mother === person.name) && child.sex === 'm'
+    ));
```

Comment:
can you use your personGender on children array? better do it that way, don't repeat yourself